packages/@aws-cdk/aws-ec2-alpha/lib/vpc-v2-base.ts

```diff
@@ -301,6 +301,12 @@ export abstract class VpcV2Base extends Resource implements IVpcV2 {
    */
   protected _internetGatewayId = '';
 
+  /**
+   * Mutable private field for the EgressOnlyInternetGatewayId
+   * @internal
+   */
+  protected _egressOnlyInternetGatewayId = '';
```

Comment:
I think this should be:
```suggestion
  protected _egressOnlyInternetGatewayId?: string';
```

The `public get egressOnlyInternetGatewayId()` method below returns `string | undefined`. The `undefined` case should be enough to signal to the caller that there is no EgressOnlyInternetGatewayId added to this Vpc. That leaves the Empty String case very ambiguous as to what it means and how it should be handled.

Same comment goes to the `_internetGatewayId` field above.